cosmwasm/contract.ts

```diff
@@ -153,113 +164,103 @@ const registerProtocol = async (
 
-    return executeContractMessage(client, config, options, 'Coordinator', msg, fee);
+    return executeContractMessage(client, config, options, 'Coordinator', msg, fee, defaultTitle, defaultDescription);
 };
 
 const registerDeployment = async (
     client: ClientManager,
     config: ConfigManager,
     options: ContractCommandOptions,
-    _args?: string[],
+    args?: string[],
     fee?: string | StdFee,
 ): Promise<void> => {
-    const { chainName } = options;
+    if (!args || args.length === 0) {
+        throw new Error('chainName is required');
+    }
+    const [chainName] = args;
     const coordinator = new CoordinatorManager(config);
     const message = coordinator.constructRegisterDeploymentMessage(chainName);
-    const msg = [JSON.stringify(message)];
+    const msg = [message];
 
-    if (!options.title || !options.description) {
-        options.title = options.title || `Register ${chainName} deployment on Coordinator`;
-        options.description = options.description || `Register ${chainName} deployment on Coordinator`;
-    }
+    const defaultTitle = `Register ${chainName} deployment on Coordinator`;
+    const defaultDescription = `Register ${chainName} deployment on Coordinator`;
 
-    return executeContractMessage(client, config, options, 'Coordinator', msg, fee);
+    return executeContractMessage(client, config, options, 'Coordinator', msg, fee, defaultTitle, defaultDescription);
 };
 
 const createRewardPools = async (
     client: ClientManager,
     config: ConfigManager,
     options: ContractCommandOptions,
-    _args?: string[],
+    args?: string[],
     fee?: string | StdFee,
 ): Promise<void> => {
-    const { chainName, epochDuration, participationThreshold, rewardsPerEpoch } = options;
-
-    if (!participationThreshold) {
-        throw new Error('Participation threshold is required');
-    }
-    if (!epochDuration) {
-        throw new Error('Epoch duration is required');
-    }
-    if (!rewardsPerEpoch) {
-        throw new Error('Rewards per epoch is required');
+    if (!args || args.length === 0) {
+        throw new Error('chainName is required');
     }
+    const [chainName] = args;
+    const { epochDuration, participationThreshold, rewardsPerEpoch } = options;
 
-    let parsedThreshold: string[];
+    let parsedThreshold;
     try {
         parsedThreshold = JSON.parse(participationThreshold);
-        if (!Array.isArray(parsedThreshold)) {
-            throw new Error('Participation threshold must be a JSON array');
-        }
-    } catch (error) {
-        throw new Error(`Invalid participation threshold format: ${error instanceof Error ? error.message : String(error)}`);
+    } catch {
+        throw new Error(`Invalid participationThreshold format. Expected JSON array, got: ${participationThreshold}`);
     }
 
+    const threshold: string[] = config.validateThreshold(parsedThreshold, '--participationThreshold');
```

Comment:
```suggestion
    const threshold: string[] = config.parseThreshold(options.participationThreshold);
```
if the goal is to parse it, just have the helper do all of it?